src/borg/repository.py

```diff
@@ -254,7 +254,13 @@ def check_transaction(self):
         index_transaction_id = self.get_index_transaction_id()
         segments_transaction_id = self.io.get_segments_transaction_id()
         if index_transaction_id is not None and segments_transaction_id is None:
-            raise self.CheckNeeded(self.path)
+            # we have a transaction id from the index, but we did not find *any*
+            # commit in the segment files (thus no segments transaction id).
```

Comment:
hmm, this is either not in 1.0.10 (see traceback in #2427) or done after the time when it already relies on the commit being there.